Backend/service-payment/src/main/java/com/bikenest/servicepayment/controllers/PaymentController.java

```diff
@@ -17,28 +32,73 @@
     PaymentService paymentService;
 
     @GetMapping("/info")
-    public String info(){
+    public String info() {
         return "Payment Service works!";
     }
 
     /**
-     * API Endpoint /payment/getclienttoken
-     * Just provide the signed JWT in the Authorization Header and then this method will create a Braintree Customer for the
-     * requester and return a temporary client token, that can be used to initiate a payment with Braintree.
+     * API Endpoint /payment/getclienttoken Just provide the signed JWT in the
+     * Authorization Header and then this method will create a Braintree Customer
+     * for the requester and return a temporary client token, that can be used to
+     * initiate a payment with Braintree.
+     * 
      * @param userInformation
      * @return
      */
     @GetMapping("/getclienttoken")
     @PreAuthorize("hasRole('USER') or hasRole('ADMIN')")
     public String getClientToken(@AuthenticationPrincipal UserInformation userInformation) throws Exception {
-        //TODO: Check if we already created a customer at braintree for this user
+        // TODO: Check if we already created a customer at braintree for this user
         boolean success = paymentService.createCustomer(userInformation.getUserId(), userInformation.getFirstName(),
                 userInformation.getLastName(), userInformation.getEmail());
         return paymentService.generateClientToken(userInformation.getUserId());
     }
 
     @GetMapping("/registerpayment")
-    public String registerPayment(){
+    public String registerPayment() {
         return "Works";
     }
+
+    @GetMapping(path = "/getIban")
+    @PreAuthorize("hasRole('USER') or hasRole('ADMIN')")
+    public ResponseEntity<PaymentResponse> getPayment(@AuthenticationPrincipal UserInformation user)
+            throws BusinessLogicException {
+        if (user.getRole() == UserRole.User || user.getRole() == UserRole.Admin) {
```

Comment:
this is not really required, because the PreAuthorize annotation will take care of it, if the WebSecurity class is configured correctly. I think you still have to edit the WebSecurity.

Reply:
should be done now